frontend/components/transaction/BidTransactionTable.vue

```diff
@@ -0,0 +1,182 @@
+            <div class="flex w-1/2 items-center space-x-2 justify-end -mr-1">
+                <div v-if="amountToBid">
+                    <button
+                        class="text-primary hover:text-primary-light whitespace-nowrap"
+                        @click="setAmountToBid(undefined)"
+                    >
+                        Reset To Total
+                    </button>
+                </div>
+                <div class="w-full flex-shrink-0">
+                    <bid-input
+                        :amount-to-bid.sync="amountToBid"
+                        :minimum-deposit-dai="minimumDepositDai"
+                        :total-price="auctionTransaction.totalPrice"
+                        :disabled="auctionTransaction.isFinished || !auctionTransaction.isActive"
+                    />
+                </div>
+            </div>
+        </div>
+        <div class="flex justify-between font-bold">
+            <div>The amount to receive</div>
+            <div>
+                <format-currency
+                    v-if="amountToReceive && isActive && !isBidAmountNaN"
+                    :value="amountToReceive"
+                    :currency="auctionTransaction.collateralSymbol"
+                />
+                <div v-else>
+                    <span class="opacity-50">UNKNOWN</span>
+                    <span class="font-bold">{{ auctionTransaction.collateralSymbol }}</span>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import BidInput from '../utils/BidInput.vue';
+import TimeTill from '~/components/common/TimeTill.vue';
+import FormatCurrency from '~/components/utils/FormatCurrency.vue';
+import PriceDropAnimation from '~/components/utils/PriceDropAnimation.vue';
+
+export default Vue.extend({
+    components: {
+        TimeTill,
+        FormatCurrency,
+        PriceDropAnimation,
+        BidInput,
+    },
+    props: {
+        auctionTransaction: {
+            type: Object as Vue.PropType<AuctionTransaction>,
+            required: true,
+        },
+        minimumDepositDai: {
+            type: Object as Vue.PropType<BigNumber>,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            amountToBid: undefined as BigNumber | undefined,
+        };
+    },
+    computed: {
+        amountToReceive(): BigNumber | undefined {
+            if (!this.amountToBid || !this.auctionTransaction.approximateUnitPrice) {
+                return this.auctionTransaction.collateralAmount;
+            }
+            return this.amountToBid.dividedBy(this.auctionTransaction.approximateUnitPrice);
+        },
+        isActive(): boolean {
+            return this.auctionTransaction.isActive && !this.auctionTransaction.isFinished;
+        },
+        isBidAmountNaN(): boolean {
+            return this.amountToBid?.isNaN() || false;
```

Comment:
Why do you need to have `|| false`? Or is it just about being explicit?

Reply:
Since `amountToBid` can be unknown it wanted an else case, but I can also force it to be a boolean via `!!this.amountToBid?.isNaN()`

Reply:
Ah, true! I think the best would be to use `this.amountToBid?.isNaN() ?? false` 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

This relatively new operator is easier to read since you know it only handles `null` or `undefined` and no weird type conversion is happening. `!!` is also fine, it's just a tip for the future!